File: TheaterSales.Device/src/Domain.ts
export interface Theater
{
  readonly id: number
  readonly name: string
}

export interface Movie
{
  readonly id: number
  readonly title: string
}

export interface Sale
{
  readonly id: number
  readonly theaterId: number
  readonly movieId: number
  readonly saleDate: string
  readonly amount: number
}

export interface TheaterSalesAggregate
{
  readonly theater: Theater
  readonly totalSales: number
}

File: FunctionalSQL.Server/src/Domain.cs
namespace FunctionalSQL.Server.Domain;

public record Theater(int Id, string Name);

public record Movie(int Id, string Title);

public record Sale(
    int Id,
    int TheaterId,
    int MovieId,
    DateOnly SaleDate,
    decimal Amount
);

public record TheaterSalesAggregate(
    Theater Theater,
    decimal TotalSales
);